src/components/CarouselSection/NewsCarousel/NewsCarousel.tsx

```diff
@@ -0,0 +1,91 @@
+"use client";
+
+import { useState, useEffect } from "react";
+import {
+  NewsWrapper,
+  NewsImage,
+  NewsTitle,
+  ButtonWrapper,
+  LoadingBar,
+  RoundButton,
+  Overlay,
+  LoadingIcon,
+} from "./NewsCarousel.styles";
+import { INews } from "@/utils/interfaces";
+
+const NewsCarousel = ({
+  newsItems,
+  loading,
+}: {
+  newsItems: INews[];
+  loading: boolean;
+}) => {
+  const [index, setIndex] = useState(0);
+  const length = 3;
+  let handler: NodeJS.Timeout;
+
+  const handleNext = (event?: { stopPropagation: () => void } | undefined) => {
```

Comment:
We could refactor this function to be like a handle update, that receives either the operation string, like "next", or the function itself